### PR TITLE
ov::Any can get value from stored string

### PR DESCRIPTION
**Pull request by apankratovantonp**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/plugins/intel_gpu/src/plugin/plugin.cpp [hunk 1]
```diff
@@ -852,19 +852,18 @@ Parameter Plugin::GetMetric(const std::string& name, const std::map<std::string,
...
+            } else if (it_streams->second.is<std::string>()) {
+                std::string n_streams_str = it_streams->second.as<std::string>();
+                if (n_streams_str != CONFIG_VALUE(GPU_THROUGHPUT_AUTO)) {
                     IE_THROW() << "[GPU_MAX_BATCH_SIZE] bad casting: GPU_THROUGHPUT_STREAMS should be either of uint32_t type or \"GPU_THROUGHPUT_AUTO\"";
                 }
+                n_streams = config.GetDefaultNStreamsForThroughputMode();
+            } else {
+                IE_THROW() << "[GPU_MAX_BATCH_SIZE] bad casting: GPU_THROUGHPUT_STREAMS should be either of uint32_t type or \"GPU_THROUGHPUT_AUTO\"";
             }
         }
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/plugin/plugin.cpp:868]: @vladimir-paramuzov please, have a look

File: src/plugins/intel_gpu/src/plugin/plugin.cpp [hunk 2]
```diff
@@ -852,19 +852,18 @@ Parameter Plugin::GetMetric(const std::string& name, const std::map<std::string,
             }
             return decltype(ov::max_batch_size)::value_type {static_cast<uint32_t>(max_batch_size)};
         }
+        auto it_streams = options.find("GPU_THROUGHPUT_STREAMS");
         if (options.find("GPU_THROUGHPUT_STREAMS") != options.end()) {
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/plugin/plugin.cpp:856]: `options.find("GPU_THROUGHPUT_STREAMS")` => `it_streams ` ?

Comment (apankratovantonp) [reply src/plugins/intel_gpu/src/plugin/plugin.cpp]: fixed

File: src/core/include/openvino/core/any.hpp [hunk 3]
```diff
@@ -621,8 +654,74 @@ class OPENVINO_API Any {
...
+                                      !std::is_same<T, std::string>::value && std::is_default_constructible<T>::value,
+                                  T>::type&
+    as() const& {
+        impl_check();
+        if (_impl->type_info() == typeid(decay_t<T>)) {
+            return *static_cast<const decay_t<T>*>(_impl->addressof());
+        } else if (_impl->type_info() == typeid(std::string)) {
+            _temp_impl = std::make_shared<Impl<decay_t<T>>>();
+            std::stringstream strm{as<std::string>()};
+            _temp_impl->read(strm);
```
Comment (ilya-lavrenov) [inline src/core/include/openvino/core/any.hpp:667]: agreed offline to make more strict contract here and convert only string which represent numbers (inference / decimal) to actual numbers, other types have fallback to current `read`

Comment (ilya-lavrenov) [reply src/core/include/openvino/core/any.hpp:667]: and let's add tests for it :)

File: src/plugins/intel_gpu/src/plugin/plugin.cpp (hunk 1 above)
Comment (sshlyapn) [reply src/plugins/intel_gpu/src/plugin/plugin.cpp:868]: Looks good, but I think I can handle this part myself in my next PR (looks like this part requires additional modification related to API 2.0), if these plugin changes is not critical for this PR

File: src/core/include/openvino/core/any.hpp (hunk 3 above)
Comment (apankratovantonp) [reply src/core/include/openvino/core/any.hpp:660]: done

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**